src/mono/mono/metadata/reflection.c

```diff
@@ -1548,8 +1548,16 @@ assembly_name_to_aname (MonoAssemblyName *assembly, char *p)
 	}
 	assembly->name = p;
 	s = p;
-	while (*p && (isalnum (*p) || *p == '.' || *p == '-' || *p == '_' || *p == '$' || *p == '@' || g_ascii_isspace (*p)))
-		p++;
+	guchar *inptr = (guchar *) p;
+	while (*p && (*p != ',')) {
+		if (quoted && (*p == '"'))
+			break;
+		guint length = g_utf8_jump_table[*inptr];
+		if (!g_utf8_validate_part (inptr, length)) {
```

Comment:
you never update `inptr` while in the loop, so this is always just looking at the first character.  I think you want `p` here.

Also this needs to check that if you advance by `length` characters that you dont' go past the end of the string (I think it can happen with a prematurely terminated utf8 input).

Reply:
`g_utf8_validate_part` is able to check if the characters of that length are within the expected range of the values. If not, it returns false. Then,`assembly_name_to_aname` will return 0.

I do need to update `inptr`. I can't use `p` here, because of type mismatch.